src/main.rs

```diff
@@ -1,50 +1,76 @@
 use anyhow::{Context, Error, Result};
-use clap::Parser;
+use clap::{Parser, Subcommand};
 use console::style;
-use serde_json::to_writer;
+use serde_json::{to_writer, Value};
 use std::fs::File;
 use std::path::PathBuf;
-use universal_sierra_compiler::compile;
+use universal_sierra_compiler::commands;
+use universal_sierra_compiler::commands::compile_contract::CompileContract;
+use universal_sierra_compiler::commands::compile_raw::CompileRaw;
 
-#[derive(Parser, Debug)]
+#[derive(Parser)]
 #[command(version)]
-struct Args {
-    /// Path to the sierra json file
-    #[arg(short, long)]
-    sierra_input_path: PathBuf,
-
-    /// Path to where casm json file will be saved
-    #[arg(short, long)]
-    casm_output_path: Option<PathBuf>,
+struct Cli {
+    #[command(subcommand)]
+    command: Commands,
+}
+
+#[derive(Subcommand)]
+enum Commands {
+    // Compile sierra of the contract
+    CompileContract(CompileContract),
+
+    // Compile sierra program (cairo_lang_sierra::program::Program)
+    CompileRaw(CompileRaw),
```

Comment:
It should be `CompileProgram`, I would consider bytecode or textual sierra a raw sierra